### ISSUE TITLE
Rule proposal: no-nested-describes

### ISSUE THREAD

**Issue by jfmengels** [lo1tuma/eslint-plugin-mocha#83]
Some consider having nested describes a bad practice, and prefer splitting tests into several files. See https://github.com/avajs/ava/issues/222 for a discussion for the introduction of nesting in `AVA` (which is currently not accepted, and probably never will).

I'm proposing this rule for people such as myself that like this train of thought but use Mocha in their projects.

#### Invalid
```js
describe('foo', function() {
  describe('bar', function() {
    // ...
  });
});
```
#### Valid
```js
describe('foo', function() {
  // ...
});

describe('bar', function() { // or even split it into several files
  // ...
});
```

Now that I think about it, another way to do this could be to only allow one `describe` per file, which would make the valid example invalid. This would probably be better as a different but complimentary rule (either allow one `describe` per file, or one `describe` per "level").

**Comment by lo1tuma** [comment]
I’m not sure if I fully understood the proposal.

Do you want to  have to rules, where one can limit to amount of nested describes and one rule that limits the usage of describes per scope?

**Comment by jfmengels** [comment]
Sorry for kind of proposing two rules at the same time and making this confusing.

My initial proposal is to have a rule that forbids having a `describe` inside of another `describe`.

And when writing this proposal, I noticed that we could take this "no nesting" (or maybe AVA-ification) one step further and forbid having more than one describe in a single file, which I think should probably be a different rule.

I proposed both at the same time as there is some overlapping (if I have nested describes, both rules would trigger). Maybe the second rule should only warn about sibling describes
```js
describe(...);
describe(...);
```
and not for nested describes
```js
describe(function() {
  describe(...);
});
```
as that one would be handled by the first rule.

tl;dr: What I'm thinking of now:
- One rule for forbidding nested describes
- One rule for forbidding sibling describes
